src/js/angular/similarity/controllers/create-index.controller.js

```diff
@@ -277,6 +280,69 @@ function CreateSimilarityIdxCtrl(
         return $translate.instant('similarity.close.btn.msg', {operation: operationType});
     }
 
+    $scope.getTabErrorMessage = (similarityQueryTab) => {
+
+        if (!$scope.similarityIndexInfo) {
+            return;
+        }
+
+        if (SimilarityQueryType.DATA === similarityQueryTab) {
+           return getSelectQueryErrorMessage();
+        }
+
+        if (SimilarityQueryType.SEARCH === similarityQueryTab) {
+            return getSearchQueryErrorMessage();
+        }
+
+        if (SimilarityQueryType.ANALOGICAL === similarityQueryTab) {
+            return getAnalogicalQueryErrorMessage();
+        }
+
+        return $translate.instant('similarity.empty.select.query.error');
+    }
+
+    const getSelectQueryErrorMessage = () => {
+        if (!$scope.similarityIndexInfo.hasSelectQuery()) {
+            return $translate.instant('similarity.empty.select.query.error');
+        }
+
+        if ($scope.similarityIndexInfo.invalidSelectQueryType) {
+            return $translate.instant('similarity.error.invalid.select.query');
+        }
+
+        if ($scope.similarityIndexInfo.invalidSelectQuery) {
+            return $translate.instant('similarity.error.query.invalid');
+        }
+    }
+
+    const getSearchQueryErrorMessage = () => {
+        if (!$scope.similarityIndexInfo.hasSearchQuery()) {
+            return $translate.instant('similarity.empty.search.query.error');
+        }
+
+        if ($scope.similarityIndexInfo.invalidSelectQueryType) {
```

Comment:
Is this correct to be `invalidSelectQueryType` or it should be `search`?

Reply:
fixed